frontend/src/components/Subscriber/Subscriber.tsx

```diff
@@ -61,13 +66,30 @@ const Subscriber = ({
     }
   }, [subscriberWrapper, isScreenShare]);
 
+  const handlePinClick = (clickEvent: MouseEvent<HTMLButtonElement>) => {
+    pinSubscriber(subscriberWrapper.id);
+    // We set hovering to false manually since onMouseLeave is not invoked when the DOM Element is moved.
+    setIsTileHovered(false);
+    // In case the DOM Element didn't move, which can happen if pinning while viewing screenshare -
+    // we use setTimeout to let the new layout render, then check if the element is still under the click event location.
+    // If so we re-enable the hover state.
+    setTimeout(() => {
+      if (subRef.current) {
+        const divRect = subRef.current.getBoundingClientRect();
+        if (isMouseEventInsideBox(clickEvent, divRect)) {
+          setIsTileHovered(true);
+        }
+      }
+    }, 0);
+  };
+
   const hasVideo = subscriberWrapper.subscriber?.stream?.hasVideo;
   const initials = subscriberWrapper.subscriber?.stream?.initials;
   const username = subscriberWrapper.subscriber?.stream?.name ?? '';
   const hasAudio = subscriberWrapper.subscriber.stream?.hasAudio;
   const audioIndicatorStyle =
     'rounded-xl absolute top-3 right-3 bg-darkGray-55 h-6 w-6 items-center justify-center flex m-auto';
-
+  const pinStyle = `flex rounded-xl absolute top-3 left-3 h-6 w-6 items-center justify-center m-auto`;
```

Comment:
Can we move this into the `PinButton` component?

Reply:
If it's forward-thinking for the participant list changes, feel free to resolve this and my other comments.

Reply:
No that's ok, we can move it now and see if it makes sense to extract later or not.